Sofa/Component/Engine/Select/src/sofa/component/engine/select/BaseROI.inl

```diff
@@ -0,0 +1,760 @@
+        WriteOnlyAccessor< Data<vector<Triangle> > > trianglesOutROI = d_trianglesOutROI;
+        WriteOnlyAccessor< Data<vector<Quad> > > quadsOutROI = d_quadsOutROI;
+        WriteOnlyAccessor< Data<vector<Tetra> > > tetrahedraOutROI = d_tetrahedraOutROI;
+        WriteOnlyAccessor< Data<vector<Hexa> > > hexahedraOutROI = d_hexahedraOutROI;
+
+        // Clear lists
+        indices.clear();
+        edgeIndices.clear();
+        triangleIndices.clear();
+        quadIndices.clear();
+        tetrahedronIndices.clear();
+        hexahedronIndices.clear();
+        indicesOut.clear();
+        edgeOutIndices.clear();
+        triangleOutIndices.clear();
+        quadOutIndices.clear();
+        tetrahedronOutIndices.clear();
+        hexahedronOutIndices.clear();
+
+        pointsInROI.clear();
+        edgesInROI.clear();
+        trianglesInROI.clear();
+        quadInROI.clear();
+        tetrahedraInROI.clear();
+        hexahedraInROI.clear();
+        pointsOutROI.clear();
+        edgesOutROI.clear();
+        trianglesOutROI.clear();
+        quadsOutROI.clear();
+        tetrahedraOutROI.clear();
+        hexahedraOutROI.clear();
+
+
+        if (d_positions.getValue().size() == 0)
+        {
+            msg_warning() << "No rest position yet defined. ROI might not work properly. \n"
+                            "This may be caused by an early initialization of the ROI before  \n"
+                            "the mesh or the MechanicalObject of the node was initialized too";
+            return;
+        }
+
+        if (!roiDoUpdate())
+        {
+            return;
+        }
+
+        // Read accessor for input topology
+        const ReadAccessor< Data<vector<Edge> > > edges = d_edges;
+        const ReadAccessor< Data<vector<Triangle> > > triangles = d_triangles;
+        const ReadAccessor< Data<vector<Tetra> > > tetrahedra = d_tetrahedra;
+        const ReadAccessor< Data<vector<Hexa> > > hexahedra = d_hexahedra;
+        const ReadAccessor< Data<vector<Quad> > > quads = d_quads;
+
+        const VecCoord& positions = d_positions.getValue();
+
+        //Points
+        for( unsigned i=0; i<positions.size(); ++i )
+        {
+            if (isPointIn(i))
+            {
+                indices.push_back(i);
+                pointsInROI.push_back(positions[i]);
+            }
+            else
+            {
+                indicesOut.push_back(i);
+                pointsOutROI.push_back(positions[i]);
+            }
+        }
+
+        auto testROI = [&](const auto& elements, const auto& predicate, const auto& strictPredicate, bool strict, 
+            auto& inIndices, auto& inROI, auto& outIndices, auto& outROI)
+            {
+                for (std::size_t i = 0; i < elements.size(); i++)
+                {
+                    const auto& e = elements[i];
+                    const bool isInROI = (strict) ? strictPredicate(e) : predicate(e);
```

Comment:
Can you not call directly `isInROI<Element>(const Element& e)` and `<Element>isInStrictROI(const Element& e)` instead of passing a verbose lambda?

Reply:
I dont think so ; I need to call isBlablaInROI, as it is virtual and can be overriden in the subsequent inheriting classes.
It is not possible to make virtual templated method unfortunately

Reply:
right, I did not see that the functions were virtual